integreat_cms/static/src/js/poi-categories/poicategory-colors-icons.ts

```diff
@@ -0,0 +1,43 @@
+const renderColorPreview = ({ target }: Event) => {
+    const colorSelect = target as HTMLSelectElement;
+    const colorPreview = document.getElementById("poi-category-color-preview");
+    const selectedColor = colorSelect.options[colorSelect.selectedIndex].value;
+    console.debug(selectedColor);
+    if (selectedColor) {
+        colorPreview.style.backgroundColor = selectedColor;
+        colorPreview.classList.remove("hidden");
+        colorSelect.classList.add("rounded-none", "rounded-r", "border-l-0");
+    } else {
+        colorPreview.classList.add("hidden");
+        colorSelect.classList.remove("rounded-none", "rounded-r", "border-l-0");
+    }
+};
+
+const renderIconPreview = ({ target }: Event) => {
+    const iconSelect = target as HTMLSelectElement;
+    const iconPreview = document.getElementById("poi-category-icon-preview");
+    const selectedIcon = iconSelect.options[iconSelect.selectedIndex].value;
+    iconPreview.setAttribute("src", selectedIcon);
+    console.debug(selectedIcon);
+
+    if (selectedIcon) {
+        iconPreview.removeAttribute("hidden");
+        iconSelect.classList.add("rounded-none", "rounded-r", "border-l-0");
+    } else {
+        iconPreview.setAttribute("hidden", "true");
+        iconSelect.classList.remove("rounded-none", "rounded-r", "border-l-0");
+    }
+};
+
+window.addEventListener("load", () => {
+    const colorSelect = document.getElementById("id_color") as HTMLSelectElement;
+    colorSelect?.addEventListener("change", renderColorPreview);
+    // Simulate initial input after page load
+    colorSelect?.dispatchEvent(new Event("change"));
+
+    const iconSelect = document.getElementById("id_icon") as HTMLSelectElement;
```

Comment:
```suggestion
    const iconSelect = document.getElementById("poi-category-icon-field") as HTMLSelectElement;
```
I noticed a conflict with the icon field:
https://github.com/digitalfabrik/integreat-cms/blob/5c6bc9787127815196bb170b69fa39ec2812fbbb/integreat_cms/static/src/js/forms/icon-field.tsx#L9

So we have to rename this one.